modules/ROOT/pages/binary-streaming.adoc

```diff
@@ -51,6 +51,48 @@ By allowing elements such as `<repeatable-file-store-stream />`, `<repeatable-in
 </file:read>
 ----
 
+== Stream from Existing Files
+
+Available since version 1.3.
+
+When the `<repeatable-file-store-stream>` strategy is used, a file is used to temporarily hold information that doesn’t fit into the in-memory buffer. There are particular cases in which this is inefficient and leads to suboptimal IO and disk space requirements.
+
+Most common cases of this include:
+
+Scenario 1::
+You are building an Operation or Source that returns the content of a file. The component that you developed opens a file and returns an InputStream that contains the file content. If the file is bigger than the memory buffer, the <repeatable-file-store-stream> starts building a temporary file with the stream content. That file becomes an exact duplicate of the original file that was already opened
+
+Scenario 2::
+There is a variation of the above in which an Operation or Source that obtains an InputStream and needs to pre-process it *before* it is returned. Because the original stream is not yet pushed to the flow, it hasn’t been made repeatable. As a result, the source needs to implement its own buffering by storing the processed contents into a temporary file as it processed the original stream. When pre-processing finishes, the source then pushes the contents of the file, going back to scenario 1.
+
+The above scenarios articulate a need to get a repeatable stream from the content of an existing file, without the repeatable streaming framework creating a new file of its own, since the contents already originate from a file.
```

Comment:
```suggestion
These two scenarios articulate a need to get a repeatable stream from the content of an existing file, without the repeatable streaming framework creating a new file of its own, since the contents already originate from a file. This prevents unnecessary duplication of the file.
```

Reply:
NTS, this might be redundant. check.